### PR TITLE
Implementation of URLSessionTask.cancel()

### PR DESCRIPTION
**Pull request by pushkarnk**

### CODE REVIEW

**Comment by pushkarnk** [comment]
@swift-ci please test

File: Foundation/NSURLSession/NSURLSessionTask.swift [hunk 1]
```diff
@@ -202,8 +202,21 @@ open class URLSessionTask : NSObject, NSCopying {
...
+        workQueue.sync {
+            guard self.state == .running || self.state == .suspended else { return }
+            self.state = .canceling
+            self.workQueue.async {
+                guard let s = self.session as? URLSession else { fatalError() }
+                guard let delegate  = s.delegate as? URLSessionTaskDelegate else { fatalError() }
+                s.delegateQueue.addOperation {
+                    delegate.urlSession(s, task: self, didCompleteWithError: NSError(domain: NSURLErrorDomain, code: NSURLErrorCancelled, userInfo: nil))
+                    self.internalState = .taskCompleted
+                }
```
Comment (naithar) [inline Foundation/NSURLSession/NSURLSessionTask.swift]: Wouldn't a call of `completeTask(withError:)` cover all situations when there is a need to perform a task completion?

**Comment by pushkarnk** [comment]
@swift-ci please test

**Comment by pushkarnk** [comment]
@naithar Thanks for the suggestion. Can you please review?

**Comment by naithar** [comment]
Looks good to me. :)

**Comment by pushkarnk** [comment]
@parkera Can you please review?  Thanks.

**Comment by parkera** [comment]
Looks reasonable to me.

**Comment by pushkarnk** [comment]
Thanks.

@swift-ci please test and merge

**Comment by pushkarnk** [comment]
@swift-ci please test and merge

**Pull request merged**